File: ios/Test/BatchSessionTest.m
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

#import "LRBatchSession.h"
#import "GroupServiceTest.h"
#import "LRGroupService_v62.h"

/**
 * @author Bruno Farache
 */
@interface BatchSessionTest : GroupServiceTest
@end

@implementation BatchSessionTest

- (void)testBatch {
	LRBatchSession *session = [[LRBatchSession alloc]
		initWithSession:self.session];

	LRGroupService_v62 *service = [[LRGroupService_v62 alloc]
		initWithSession:session];

	NSError *error;
	[service getUserSites:&error];
	[service getUserSites:&error];

	NSArray *commands = session.commands;
	XCTAssertEqual([commands count], 2);

	NSArray *result = [session invoke:&error];

	commands = session.commands;
	XCTAssertEqual([commands count], 0);

	[self assert:result[0] error:error];
	[self assert:result[1] error:error];

	result = [session invoke:&error];

	XCTAssertNil(result);
}

@end

File: android/services-v62/src/test/java/com/liferay/mobile/android/BatchSessionTest.java
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.mobile.android;

import com.liferay.mobile.android.service.BatchSessionImpl;
import com.liferay.mobile.android.v7.group.GroupService;

import java.io.IOException;

import org.json.JSONArray;

import org.junit.Test;

import static org.junit.Assert.*;

/**
 * @author Bruno Farache
 */
public class BatchSessionTest extends BaseTest {

	public BatchSessionTest() throws IOException {
		super();
	}

	@Test
	public void getUserSites() throws Exception {
		BatchSessionImpl session = new BatchSessionImpl(this.session);

		GroupService service = new GroupService(session);

		service.getUserSitesGroups();
		service.getUserSitesGroups();

		JSONArray result = session.invoke();

		GroupServiceTest.assertUserSites(result.getJSONArray(0));
		GroupServiceTest.assertUserSites(result.getJSONArray(1));

		result = session.invoke();

		assertNull(result);
	}

}